mne/viz/tests/test_ui_events.py

```diff
@@ -0,0 +1,271 @@
+# Authors: Marijn van Vliet <w.m.vanvliet@gmail.com>
+#
+# License: Simplified BSD
+import matplotlib.pyplot as plt
+import pytest
+
+from mne.datasets import testing
+from mne.viz import ui_events
+
+subjects_dir = testing.data_path(download=False) / "subjects"
+
+
+@pytest.fixture
+def event_channels():
+    """Fixture that makes sure each test starts with a fresh UI event chans dict."""
+    ui_events._event_channels.clear()
+    return ui_events._event_channels
+
+
+@pytest.fixture
+def event_channel_links():
+    """Fixture that makes sure each test starts with a fresh channel links dict."""
+    ui_events._event_channel_links.clear()
+    return ui_events._event_channel_links
+
+
+@pytest.fixture
+def disabled_event_channels():
+    """Fixture that makes sure each test starts with a fresh disabled channels set."""
+    ui_events._disabled_event_channels.clear()
+    return ui_events._disabled_event_channels
+
+
+@testing.requires_testing_data
+def test_get_event_channel(event_channels):
+    """Test creating and obtaining a figure's UI event channel."""
+    # At first, no event channels exist
+    assert len(event_channels) == 0
+
+    # Open a figure and get the event channel. This should create it.
+    fig = plt.figure()
+    ui_events._get_event_channel(fig)
+    assert len(event_channels) == 1
+    assert fig in event_channels
+
+    # Closing a figure should delete the event channel.
+    # During tests, matplotlib does not open an actual window so we need to force the
+    # close event.
+    fig.canvas.callbacks.process("close_event", None)
+    assert len(event_channels) == 0
+
+    # TODO: Different types of figures: Brain, MNEFigure, Figure3D
+
+
+def test_publish(event_channels):
+    """Test publishing UI events."""
+    fig = plt.figure()
+    ui_events.publish(fig, ui_events.TimeChange(time=10.2))
+
+    # Publishing the event should have created the needed channel.
+    assert len(event_channels) == 1
+    assert fig in event_channels
+
+
+def test_subscribe(event_channels):
+    """Test subscribing to UI events."""
+    global callback_called
+    callback_called = False
+
+    def callback(event):
+        """Respond to time change event."""
+        global callback_called
+        callback_called = True
+        assert isinstance(event, ui_events.TimeChange)
+        assert event.time == 10.2
```

Comment:
FYI to avoid this global business another (quicker/easier usually) approach is to use something mutable like a list:
```
callback_calls = list()

def callback(event):
    callback_calls.append(event)

...
assert len(callback_calls)  # or whatever
```
It automatically deals with this scoping stuff. I'll push a commit to do it then mark for merge when green, but if you really like the `global` approach I can live with it (and you can put it back in the next PR)!

Reply:
No strong feelings about it :)